Editor/Module Management/ModuleInstaller.cs

```diff
@@ -1,64 +1,47 @@
 using System;
-using System.IO;
 using System.Linq;
 using UnityEditor;
 using UnityEngine;
 using System.Threading;
 using UnityEditor.PackageManager;
 
-
 namespace ReadyPlayerMe.Core.Editor
 {
-    [InitializeOnLoad]
-    public class ModuleInstaller 
+    public static class ModuleInstaller 
     {
         private const string PROGRESS_BAR_TITLE = "Ready Player Me";
 
         public static Action ModuleInstallComplete;
 
 #if !DISABLE_AUTO_INSTALLER
-        static ModuleInstaller()
+        public static void Init()
         {
             var listRequest = Client.List(true);
             while (!listRequest.IsCompleted)
-                Thread.Sleep(100);
+                Thread.Sleep(20);
             if (HasAnyMissingModule())
             {
-                EditorApplication.update += InstallModules;
```

Comment:
need to really test this change as I had it in combination with the EditorApplication.Update because there was some issues where this would run before the assets were ready (mostly related to the Scriptable Objects creation). it was causing null references. 
It might be fine now though with all the latest changes but just mentioning here for awareness